libs/designer/src/lib/core/queries/connector.ts

```diff
@@ -55,7 +55,7 @@ export const getListDynamicValues = async (
       (connectionId ?? '').toLowerCase(),
       connectorId.toLowerCase(),
       operationId.toLowerCase(),
-      getParametersKey(parameters).toLowerCase(),
+      parameterAlias,
```

Comment:
You are assuming here every parameter has parameterAlias, only openApi connectors may have parameterAlias, parametersKey should be unique for parameters even if they have an alias. If you are not getting the key then there is a bug somewhere in that code.
Key is supposed to be unique.

Reply:
Can you give me an example of such manifest? Point me to your connector manifest and it should be debugged. We have manifests (non open api) which have multiple dynamic values and they do work... parameterKey empty meaning the code which generates parameter key is buggy, your scenario fix should not be here

Reply:
![image](https://user-images.githubusercontent.com/91710207/229936885-cc2a43c9-19a4-4815-a576-ac1aa946e8ee.png)
I'm not sure if this is what you're asking for.

Reply:
I think the issue here might be arising from where the parameter key is expected to be found. In our `inputsDefinition` block, we do not have a `parameterKey`, and `parameters` is just an empty object:

![image](https://user-images.githubusercontent.com/1350074/232894081-15265d35-53e8-44fd-a164-a995625fc475.png)

This means that the local scope from `getListDynamicValues` looks like the following, and produces an empty parameter key:

![image](https://user-images.githubusercontent.com/1350074/232894134-9f438fc1-07a2-4d79-a621-020ff0850c87.png)

It looks like our parameter keys are stored in `nodeInputs`, which I don't think is being considered here:

![image](https://user-images.githubusercontent.com/1350074/232894007-bd0c4119-19c7-49ce-bc03-b72f78a13ac5.png)

In fact, I can indeed see the parameter keys properly set in the redux state:

![image](https://user-images.githubusercontent.com/1350074/232894382-25040619-9eeb-4dd7-8881-e0b88a6ce7f0.png)

So it looks like the `parameters` being included here might not be considering those from `nodeInputs` in OpenAPI. Is this something we need to fix in the OpenAPI deserializer?

Reply:
Digging a bit further, it looks like `getDynamicValues` wants to get the `parameters` value from `definition.extension`, but we don't use that for OpenAPI.

![image](https://user-images.githubusercontent.com/1350074/232897872-31f2e3f1-fe38-44fa-8115-f64be23d46d7.png)

![image](https://user-images.githubusercontent.com/1350074/232897949-97aa3cfe-bb10-472d-9794-966e9fba1308.png)

This might actually be on our side. I'll take a deeper look.

Reply:
I think I figured this one out.

The way this works is that we use the `dependencies.inputs` object to retrieve `dependencyInfo`. However, this object only contains information about the parameter that it depends on, not the parameter itself.

![image](https://user-images.githubusercontent.com/1350074/232903024-e5caa83f-f178-48b8-a065-c60de0d1b299.png)

![image](https://user-images.githubusercontent.com/1350074/232903282-8c658af6-c8fe-4963-87e5-23bc884a6eeb.png)

This means that any input which has no dependencies will have a blank parameter key. Unless non-OpenAPI manifests have some other way of handling `dependencies.inputs`, I expect all manifests will see the same issue, where input parameter keys are blank.

So this does indeed look like a bug in LAUX. Can we confirm that this exists for manifest operations in LA Portal today?

cc @preetriti1

Reply:
No, to fetch dynamic content why would you need the parameter key of the parameter on which dynamic call is being made, at that point all you need is connector information, the operationId (the operation making the dynamic call like GetAllTeams) and the dependent parameters.
This is how it works in backend, unless you guys are going something with the parameter key of the parameter on which dynamic calls needs to be made.

Reply:
As discussed offline, merging the parameters for dynamicState and inputs to fetch the right parametersKey to make QueryCacheKey distinct